### DISCUSSION TITLE
[Question]: How to detect if object is inbound or outbound relative to the camera?

### DISCUSSION THREAD

**Discussion by glossyio** [blakeblackshear/frigate#9901] [Ask A Question] [answered]
### What is your question:

I would like to be able to attach an attribute (i.e. as a sub_label to later query) to an event that tells me whether the detected object was inbound (coming towards the camera) or outbound (going away from the camera).  Has anyone put work or thought into this that I can leverage?
- How difficult of a problem is this to solve for with the current Frigate capabilities?  
- What would the "best practice" be in terms of accuracy and simplicity of implementation?

Some thoughts I had:
1. Set up a **beginning/ending zone** (for close to camera and far away from camera); e.g. to see if the object first entered close zone then far zone for an _outgoing_ object.  I could calculate this via the  `frigate/events` [MQTT message](https://deploy-preview-6262--frigate-docs.netlify.app/integrations/mqtt/#frigateevents), specifically with the `"current_zones": ["driveway"]` and `"entered_zones": ["yard", "driveway"],`, looking for the final message with `end_time` and seeing if the far box in `current_zones` represents near or far.
2. See if the **bounding box got larger or smaller** using the `frigate/events` attribute `box` for the same `id` with the `before` and `after` keys, or type `new` and `end`?

Love to hear if someone else had done anything around this.  Thank you!

FYI - I'm using a [node-red](https://nodered.org/) flow compute the logic, with an MQTT connection to Frigate (v 0.12.1) and API call to attach the sub_label to an event.

**Comment by NickM-27** [comment, marked as answer]
zones would be the easiest way to do this. The mqtt entered_zones keeps them in order

**Reply by glossyio** [reply]
Oh man, `entered_zones` is _in order_??  That seems like a no-brainer to use. Aside from the manual zones setup, it has to be the best way to do it. Thank you!

**Comment by nwish** [comment]
I’ve been thinking about doing something similar with my cameras that can see our street to track eastbound vs westbound traffic.  No particular reason, just for fun lol

**Reply by glossyio** [reply]
Yeah, that's exactly my use case. I'm counting cars, bikes, pedestrians. 
Long story: I have a [OPS243 Doppler radar](https://omnipresense.com/product/ops243-doppler-radar-sensor/) for speed detection. It reports back inbound-outbound, but I need a way to attach that to the event, in case there are 2 cars each direction at the same time.

**Reply by nwish** [reply]
You sure you’re not catching speeders and forwarding the plates to the local police? Haha!!  You could ping the radar unit as soon as one of your target objects is detected by Frigate, and attach that as a sub label, but you may not always get the correct moving object, but I think you’ve already thought of that.
